tests/conftest.py

```diff
@@ -2,6 +2,19 @@
 from typing import Any
 
 import pytest
+from typer.testing import CliRunner
+
+from deploy_tools.__main__ import app
+
+runner = CliRunner()
+
+
+def run_cli(*args):
+    result = runner.invoke(app, [str(x) for x in args])
+    if result.exception:
+        raise result.exception
+    assert result.exit_code == 0, result
```

Comment:
Does this properly raise the original exception with original traceback? Or does it produce an exception which appears to only have been raised here, potentially confusing things?

Reply:
It did show the original traceback when I was debugging the test so I think it's all good. I'm not clear why it works but I think I originally got this from the typer docs.

Reply:
Not sure what is happening here but I answered several of these comments before and they are not showing in the conversation. Meh.

When I was debugging the test I did get full traceback so this works fine. I'm not sure how it works but I believe this approach is documented in `typer`